connect-examples/oauth/java/src/main/java/com/squareup/oauthexample/OAuthHandler.java

```diff
@@ -117,39 +127,48 @@ public void handle(HttpExchange t) throws IOException {
         return;
       }
 
-      ObtainTokenRequest body = new ObtainTokenRequest();
-      body.setClientId(APPLICATION_ID);
-      body.setClientSecret(APPLICATION_SECRET);
-      body.setCode(authorizationCode);
-      body.setGrantType("authorization_code");
-
-      //The default base path is 'https://connect.squareup.com'
-      //When you are testing OAuth in the Square Sandbox, this resets
-      //the base path to the Square Sandbox domain
-      ApiClient apiClient = Configuration.getDefaultApiClient();
-      apiClient.setBasePath(CONNECT_HOST);
-
-      OAuthApi oAuthApi = new OAuthApi();
-      ObtainTokenResponse response = null;
-      try {
-        response = oAuthApi.obtainToken(body);
-      } catch (ApiException e) {
-        System.out.print(e.getResponseBody());
-        t.sendResponseHeaders(405, 0);
-        t.getResponseBody().close();
-        return;
-      }
+      SquareClient client = new SquareClient.Builder()
+            .environment(ENVIRONMENT)
+            .build();
 
-      if (response != null) {
-        System.out.println("Access token: " + response.getAccessToken());
-        System.out.println("Refresh token: " + response.getRefreshToken());
-        System.out.println("Authorization succeeded!");
-        t.sendResponseHeaders(200, 0);
-        t.getResponseBody().close();
+      List<String> bodyScopes = new LinkedList<>();
+      for (String scope : SCOPES) {
+        bodyScopes.add(scope);
       }
 
-      t.sendResponseHeaders(200, 0);
-      t.getResponseBody().close();
+      ObtainTokenRequest body = new ObtainTokenRequest.Builder(
+        APPLICATION_ID,
+        APPLICATION_SECRET,
+        "authorization_code")
+      .code(authorizationCode)
+      .scopes(bodyScopes)
+      .redirectUri("http://localhost:8000/callback")
+      .build();
+
+      OAuthApi oAuthApi = client.getOAuthApi();
+
+      oAuthApi.obtainTokenAsync(body).thenAccept(result -> {     
+        if (result != null) {
+          System.out.println("Access token: " + result.getAccessToken());
+          System.out.println("Refresh token: " + result.getRefreshToken());
+          System.out.println("Merchant id: " + result.getMerchantId());
+          System.out.println("Authorization succeeded!");
+          try {
+            t.sendResponseHeaders(200, 0);
+            t.getResponseBody().close();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+        }
+      }).exceptionally(exception -> {
+        try {
+          t.sendResponseHeaders(405, 0);
+          t.getResponseBody().close();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+        return null;
+      });
```

Comment:
Could you add a few additional comments in this block?

Reply:
done